addons/website/models/res_users.py

```diff
@@ -26,6 +25,20 @@ def _has_unsplash_key_rights(self):
             return True
         return super(ResUsers, self)._has_unsplash_key_rights()
 
+    @api.constrains('login', 'website_id')
+    def _check_login(self):
+        """ Do not allow two users with the same login without website """
+        self.flush(['login', 'website_id'])
+        self.env.cr.execute(""" SELECT login
+            FROM res_users
+            WHERE login IN (SELECT login FROM res_users WHERE id IN %s AND website_id IS NULL)
+                AND website_id IS NULL
+            GROUP BY login
+            HAVING COUNT(*) > 1
+        """, (tuple(self.ids),))
```

Comment:
* probably align SQL words
* Blackify this
```
self.env.cr.execute(
    """SELECT login
         FROM res_users
        WHERE login IN (SELECT login FROM res_users WHERE id IN %s AND website_id IS NULL)
          AND website_id IS NULL
     GROUP BY login
       HAVING COUNT(*) > 1
    """,
    (tuple(self.ids),),
)
```

Reply:
The subquery is actually not needed.
```python
self.env.cr.execute(
    """SELECT 1
         FROM res_users
        WHERE id IN %s
          AND website_id IS NULL
     GROUP BY login
       HAVING COUNT(*) > 1
    """,
    [tuple(self.ids)],
)
```

Reply:
@nseinlet The code have been formatted.